### PR TITLE
Reorganise repo to being into line with LST canonical structure

### PR DESCRIPTION
**Pull request by rossmiller**
These changes attempt to bring the Trade Remedies API codebase into line with the evolving standard used in some other Live Services project.

**What has changed**
I changed the way that local development should be carried, the developer should now run the project from the trade-remedies-docker project which includes a makefile to allow this.

Therefore, this project includes a very lightweight make file, solely used for generating project requirements.

Print statements have been replaced with appropriate logging.

The docker-compose file simplified has been amended to rely on the docker-compose file in trade-remedies-docker project.

The Dockerfile has been simplified and the reliance on .sh files has been removed.

CircleCI now uses a much simpler process without additional containers. 

I have introduced a RUN_ASYNC setting which allows the project to be run with or without Celery. This has been introduced because in order for the tests to work on CircleCI, the project needs to be able to work without _delay_ functions.

The local.env file has been updated to correct the values and remove cruft.

A new approach has been added for adding requirements, using pip-compile.

The settings files have been amended to bring them into line with how others services consume VCAP and to remove inconsistencies and cruft.

**Checklist of prerequisites**
This branch should be used in conjunction with:
https://github.com/uktrade/trade-remedies-docker

**How to use the feature**
This feature effects local development only and has no changes on project features. It should be used with https://github.com/uktrade/trade-remedies-docker (see the readme of that project for more details).

**How the design has changed**
No changes.

**Additional notes** 
These changes were testing by running the amended codebase in conjunction with the trade-remedies-docker project and checking that the changes were non-destructive.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,75 +1,12 @@
-SHELL := /bin/bash
```
Comment (jwgulley) [inline Makefile:1]: Are we still able to run the code quality checks locally?

File: README.md [hunk 2]
```diff
@@ -9,168 +9,39 @@ Back end API to support the Trade Remedies service.
...
+## Development
 
-```
-make docker_code_quality
-```
-runs black and flake8 in a container the same as circle ci
+#### Set up
 
-#### Running locally without docker
+Firstly, you should copy local.env.example to local.env and add the necessary environment variables for a local development environment.  local.env is in .gitignore and should not be committed to the repo.
```
Comment (jwgulley) [inline README.md:16]: How will people know how to modify local.env?

File: README.md [hunk 3]
```diff
@@ -9,168 +9,39 @@ Back end API to support the Trade Remedies service.
...
 
+The project for running ClamAV locally can be found here: https://github.com/uktrade/dit-clamav-rest
 
-## Deployment
+GOV.UK notify: https://www.notifications.service.gov.uk/
 
-Trade Remedies API configuration is performed via the following environment variables:
+Companies House API: https://developer.company-information.service.gov.uk/api/docs/
 
+If you are not sure what to use for one of the values above, ask a colleague or contact the SRE team.
```
Comment (jwgulley) [inline README.md:40]: There is clearly scope to make this more repeatable.  Whether it is worth it depends on how many people will go through the process.  A possible step forward would be to put a sample file in a Jira page or similar and refer to it here.

File: docker-compose.yml [hunk 4]
```diff
@@ -1,60 +1,37 @@
-version: "2"
+version: "3"
```
Comment (jwgulley) [inline docker-compose.yml:1]: I noted that the config.yml version changes from 2 to 2.1.  Here it changes from 2 to 3.  Is consistency needed?

File: requirements/base.txt [hunk 5]
```diff
@@ -0,0 +1,88 @@
+#
+# This file is autogenerated by pip-compile
```
Comment (jwgulley) [inline requirements/base.txt:2]: If the file is autogenerated by pip-compile, perhaps consider removing it from source control ?

File: requirements/dev.txt [hunk 6]
```diff
@@ -0,0 +1,124 @@
+#
+# This file is autogenerated by pip-compile
```
Comment (jwgulley) [inline requirements/dev.txt:2]: See above comment.

File: requirements/prod.txt [hunk 7]
```diff
@@ -0,0 +1,91 @@
+#
+# This file is autogenerated by pip-compile
```
Comment (jwgulley) [inline requirements/prod.txt:2]: See above comment.

File: trade_remedies_api/audit/tasks.py [hunk 8]
```diff
@@ -39,7 +43,7 @@ def check_notify_send_status():
     ORDER BY a.created_at DESC
     """
     audits = Audit.objects.raw(SQL)
-    print(f"Total pending audits: {len(audits)}")
+    logger.debug(f"Total pending audits: {len(audits)}")
```
Comment (jwgulley) [inline trade_remedies_api/audit/tasks.py:46]: Why did you decide to make some logging statements "debug" and others "info"?

File: trade_remedies_api/cases/management/commands/reset_case_data.py [hunk 9]
```diff
@@ -75,4 +79,4 @@ def handle(self, *args, **options):
             | Q(email=os.environ.get("HEALTH_CHECK_USER_EMAIL"))
             | Q(groups__name__in=SECURITY_GROUPS_TRA)
         ).delete()
-        print("CLEAR")
+        logger.info("Completed reset_case_data command")
```
Comment (jwgulley) [inline trade_remedies_api/cases/management/commands/reset_case_data.py:82]: Always good to see log statements which are CLEAR :-)

File: trade_remedies_api/cases/tasks.py [hunk 10]
```diff
@@ -48,11 +52,11 @@ def check_measure_expiry():
     for case in cases:
         latest_expiry = case.get_state_key("LATEST_MEASURE_EXPIRY")
         if latest_expiry and latest_expiry.value:
-            print(
-                timezone.now().date() <= parse(latest_expiry.value).date(),
-                timezone.now().date(),
-                parse(latest_expiry.value).date(),
-            )
+            # print( TODO - remove if not needed
```
Comment (jwgulley) [inline trade_remedies_api/cases/tasks.py:55]: If in doubt, log it!?!

File: trade_remedies_api/cases/tests/test_case.py [hunk 11]
```diff
@@ -36,7 +36,7 @@ def get_case_fixtures(*extra):
 
 
 def load_system_params():
-    path = f"{settings.SITE_ROOT}/trade_remedies_api/core/system/parameters.json"
+    path = "core/system/parameters.json"
```
Comment (jwgulley) [inline trade_remedies_api/cases/tests/test_case.py:39]: Are there any possible disadvantages to removing the reference to settings.SITE_ROOT?

File: trade_remedies_api/core/base.py [hunk 12]
```diff
@@ -277,6 +278,4 @@ def to_json(self, fields, obj=None, context=None, *args, **kwargs):
                 out[field] = val
             except AttributeError:
                 pass
-                # if field:
-                #    print(f'Field not found "{field}"')
```
Comment (jwgulley) [inline trade_remedies_api/core/base.py:281]: Not worth a log message?

File: trade_remedies_api/core/management/commands/load_sysparams.py [hunk 13]
```diff
@@ -25,18 +28,18 @@ def add_arguments(self, parser):
         parser.add_argument(
             "--path",
             type=str,
-            default=f"{settings.SITE_ROOT}/trade_remedies_api/core/system/parameters.json",
+            default="core/system/parameters.json",
```
Comment (jwgulley) [inline trade_remedies_api/core/management/commands/load_sysparams.py:31]: See similar comment above.

File: trade_remedies_api/core/tasks.py [hunk 14]
```diff
@@ -33,7 +36,12 @@ def send_mail(email, context, template_id, reference=None, audit_kwargs=None):
     if audit_kwargs:
         audit_kwargs.setdefault("audit_type", AUDIT_TYPE_NOTIFY)
         audit_kwargs = new_audit_record_to_dict(**audit_kwargs)
-    send_mail_task.delay(email, context, template_id, reference, audit_kwargs)
+
+    if settings.RUN_ASYNC:
+        send_mail_task.delay(email, context, template_id, reference, audit_kwargs)
+    else:
+        send_mail_task(email, context, template_id, reference, audit_kwargs)
+
```
Comment (jwgulley) [inline trade_remedies_api/core/tasks.py:44]: Is this a fix to an issue with email being sent?  Presumably this issue does not arise in production?

File: trade_remedies_api/documents/tasks.py [hunk 15]
```diff
@@ -36,7 +39,7 @@ def index_document(document_id, case_id=None):
 
     document = Document.objects.get(id=document_id)
     result = document.elastic_index(case=case_id)
-    print(result)
+    logger.info(result)
```
Comment (jwgulley) [inline trade_remedies_api/documents/tasks.py:42]: Does the result need a string to say what it is?

File: trade_remedies_api/workflow/models.py [hunk 16]
```diff
@@ -243,7 +248,7 @@ def evaluate_outcome(self, node_key, **kwargs):
                     outcome = Outcome(node, self, spec=spec, **kwargs)
                     result = outcome.execute()
         else:
-            print(f"NO outcome spec: {node.get('outcome_spec')}")
+            logger.debug(f"NO outcome spec: {node.get('outcome_spec')}")
```
Comment (jwgulley) [inline trade_remedies_api/workflow/models.py:251]: Why debug?  If this is a non standard execution path, do we not to know it happened?

**Review by jwgulley — COMMENTED** [review]
Bravo, Ross.  Looks good to me and I have been using the result successfully for development.  Obviously, I don't understand all the details, but I hope some of the comments below are helpful.  If not, please just close the comment down.

**Review by markhigham — CHANGES_REQUESTED** [review]
The api container fails to start locally or on circleci
django.core.exceptions.ImproperlyConfigured: Set the DATABASE_URL environment variable

**Review by markhigham — APPROVED** [review]

**Pull request merged**